Added information that Iterable perform check on phone number field (#4340)
segmentio/segment-docs: src/connections/destinations/catalog/iterable/index.md

Iterable keys users by `email` or a user ID. This user ID will be the Segment `userId` if sent. To use a Segment `userId` for identify calls, first call identify with both a `userId` and `email`. Iterable won't accept the request and throws an error if you fail to send one of either the `userId` or `email`.

> If you send `phone` in traits, Iterable will perform some checks on the phone number before showing in User Profiles. You can read more about Iterable's phone number field here: https://support.iterable.com/hc/en-us/articles/211970843-SMS-Overview-#contact-phone-numbers.

Reviewer: ```suggestion
If you send `phone` in traits, Iterable performs checks on the phone number before showing them User Profiles. Read more about [Iterable's phone number field](https://support.iterable.com/hc/en-us/articles/211970843-SMS-Overview-#contact-phone-numbers){:target="_blank"}.
```
